src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs

```diff
@@ -117,9 +119,18 @@ private void StartServiceCallback(object sender, EventArgs e)
 
                 _serviceChannel.StartListening(null);
 
-                var componentModel = ServiceProvider.GetService<SComponentModel, IComponentModel>();
-                var asyncCompletionTracker = componentModel.GetService<AsyncCompletionTracker>();
-                asyncCompletionTracker.StartListening();
+                // Async initialization is a workaround for deadlock loading ExtensionManagerPackage prior to
+                // https://devdiv.visualstudio.com/DevDiv/_git/VSExtensibility/pullrequest/381506
+                _ = ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    await TaskScheduler.Default;
+
+                    var componentModel = await AsyncServiceProvider.GlobalProvider.GetServiceAsync<SComponentModel, IComponentModel>().ConfigureAwait(false);
+                    Assumes.Present(componentModel);
+
+                    var asyncCompletionTracker = componentModel.GetService<AsyncCompletionTracker>();
+                    asyncCompletionTracker.StartListening();
```

Comment:
What ensures the rest of this will run before it's needed for a later test?

Reply:
Nothing does, but it's also only used by the old integration test harness